### DISCUSSION TITLE
Limit switches on Eleksmaker A3 pro

### DISCUSSION THREAD

**Discussion by AndyS1957** [arkypita/LaserGRBL#1708] [Q&A]
Hello everybody,

I'm Andy from Germany.
I'm using Laser GRBL on my Eleksmaker A3 pro since 2-3 years and i am very happy with both. Now i want to install XY limit switches on the Eleksmaker, but i don't know how to connect them.
The controller board is a Mana SE. It has a 14pin connector, where i can find the Arduino pins D9 (X) and D10(Y) for the limit switches. So far, so good. But how to connect them? Have the switches to be "normal open" or "normal closed"? Do i need pull up resistors an capacitors for noice reduction? Or ist it better to use opto couplers? And ist the custom firmware "v1.1h XY Homing [30.08.2019]" the right firmware? Which GRBL settings do i have to set in LaserGRBL?
Well, lots of questions...! I would be happy, if i get some answeres! Thank you for that!
By the way, i'm using
- Windows 7 64bit on an old Laptop
- LaserGRBl 4.6.0
- GRBL 1.1h

Best regards
Andy

**Comment by gmmanonymus111** [comment]
Sorry for the late reply, [here](https://github.com/gnea/grbl/wiki/Wiring-Limit-Switches)'s same info about limit switches wiring!

I'd rather follow the [instructions provided by Eleksmaker](https://wiki.eleksmaker.com/doku.php?id=extra_module) instead of flashing the firmware from LaserGRBL. (Not that it is that bad, but 3rd party manufacturers might switch some pins, add/remove some features etc... Note: If you decide to try LaserGRBL's firmware, it is advised to make/get a backup firmware just in case something goes wrong.)

**Reply by AndyS1957** [reply]
Thank you very much!
